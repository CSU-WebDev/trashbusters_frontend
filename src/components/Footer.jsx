import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

export default function Footer() {
  const currentYear = new Date().getFullYear();

  return (
    <footer id='footer'>
      <Container fluid>
        <Row>
          <Col className="text-center">
            <p className='mb-1'>Copyright &copy; {currentYear}</p>
            <p>Footer</p>
          </Col>
        </Row>
      </Container>
    </footer>
  );
}
