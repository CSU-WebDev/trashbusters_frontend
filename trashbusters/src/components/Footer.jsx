import React from 'react'
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

export default function Footer() {
    const currentYear = new Date().getFullYear();
    return (
        <>
            <p className='mb-1'>Copyright &copy; {currentYear}</p>
            <p> Footer</p>
        </>
    )
}