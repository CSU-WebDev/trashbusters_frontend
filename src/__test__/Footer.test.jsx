import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from '../components/Footer';

describe('Footer component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Footer with correct date', () => {
    render(<Footer />);
    const footerContent = screen.getByTestId('footer');
    const currentDate = new Date().toLocaleString('default', { month: 'long', year: 'numeric' });
    expect(footerContent).toHaveTextContent(
        currentDate
    );
  });

});

  