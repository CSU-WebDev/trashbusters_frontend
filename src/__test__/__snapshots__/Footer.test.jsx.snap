// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    data-testid="footer"
    id="footer"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <p>
            Trashbusters Â© May 2023 
          </p>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
